sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -569,6 +575,40 @@ public static void SetTraceState(this Activity activity, string? tracestate)
             SetTraceStateStringMethod(activity, tracestate);
         }
 
+        public static void SetErrorStatus(this Activity activity, string description)
+        {
+            if (SetErrorStatusMethod == null)
+            {
+                if (ActivityStatusCodeType == null)
+                {
+                    SetErrorStatusMethod = (_, _, _) => { };
```

Comment:
We set it to a no-op so to avoid using reflection whenever the method is called to attempt to look it up, i.e. we have 3 states - null (before reflection is used for the first time), no-op (reflection was used but we don't have the method on this version), filled out method (we have the method on this version).

Reply:
That's fair - I suppose a nullable bool could accomplish the same thing. However, I'm hesitant to change this as we use the same pattern throughout, and these are static lambdas so perf impact should be minimal.

Reply:
Cool, yeah, I am fine with whatever you decide.